src/sentry/sentry_metrics/querying/data_v2/execution.py

```diff
@@ -99,160 +96,223 @@ def _build_aligned_seq(
+                replace(updated_metrics_query.rollup, orderby=self.order.to_snuba_order())
+            )
+
+        if self.limit:
+            updated_metrics_query = updated_metrics_query.set_limit(self.limit)
+        else:
+            updated_metrics_query = updated_metrics_query.set_limit(SNUBA_QUERY_LIMIT)
+
+        return updated_metrics_query
+
+    def is_empty(self) -> bool:
+        return not self.metrics_query.scope.org_ids or not self.metrics_query.scope.project_ids
+
+    @classmethod
+    def _filter_blocked_projects(
+        cls,
+        metrics_query: MetricsQuery,
         organization: Organization,
-        projects: set[Project],
+        projects: Sequence[Project],
         blocked_metrics_for_projects: Mapping[str, set[int]],
-    ) -> "ExecutableQuery":
-        """
-        Returns a new `ExecutableQuery` with the projects for which all the queries are not blocked. In case no projects
-        exist, the query will be returned with empty projects, signaling the executor to not run the query.
-        """
+    ) -> MetricsQuery:
         intersected_projects: set[int] = {project.id for project in projects}
 
-        for queried_metric in QueriedMetricsVisitor().visit(self.metrics_query.query):
+        for queried_metric in QueriedMetricsVisitor().visit(metrics_query.query):
             blocked_for_projects = blocked_metrics_for_projects.get(queried_metric)
             if blocked_for_projects:
                 metrics.incr(key="ddm.metrics_api.blocked_metric_queried", amount=1)
                 intersected_projects -= blocked_for_projects
 
-        return replace(
-            self,
-            metrics_query=self.metrics_query.set_scope(
-                MetricsScope(
-                    org_ids=[organization.id],
-                    project_ids=list(intersected_projects),
-                )
-            ),
+        return metrics_query.set_scope(
+            MetricsScope(
+                org_ids=[organization.id],
+                project_ids=list(intersected_projects),
+            )
         )
 
+    @classmethod
+    def _align_date_range(cls, metrics_query: MetricsQuery) -> MetricsQuery:
+        # We use as a reference the interval supplied via the initial version of the query.
+        interval = metrics_query.rollup.interval
+        if interval:
+            modified_start, modified_end, _ = to_intervals(
+                metrics_query.start,
+                metrics_query.end,
+                interval,
+            )
+            if modified_start and modified_end:
+                return metrics_query.set_start(modified_start).set_end(modified_end)
+
+        return metrics_query
+
 
 @dataclass(frozen=True)
 class QueryResult:
-    series_executable_query: ExecutableQuery | None
-    totals_executable_query: ExecutableQuery | None
+    series_executable_query: MetricsQuery | None
+    totals_executable_query: MetricsQuery | None
```

Comment:
Decided to keep the query directly, since there was no real benefit of keeping the original `ExecutableQuery` because you can obtain the same information by inspecting the `MetricsQuery` itself.